pkg/server/server.go

```diff
@@ -540,31 +550,31 @@ func textifyJobTitles(jobs []*database.JobPost) string {
 }
 
 func (s Server) RenderPageForDeveloperRegistration(w http.ResponseWriter, r *http.Request, htmlView string) {
-	topDevelopers, err := database.GetTopDevelopers(s.Conn, 10)
+	topDevelopers, err := s.DeveloperRepo.GetTopDevelopers(10)
```

Comment:
We can still keep this here and inject repositories into the server for the time being. But the idea and direction of the refactoring is at some point to remove as much logic as possible from the server and have composable packages that can be tested and injected as needed directly into the handlers if that makes sense. For now the work you are doing on repositories is already great!

Reply:
I'm still struggling here, I had to pass devRepo 2 levels down and tbh couldn't figure a way to solve it in a better way, it feels weird

Reply:
This is what is troublesome and confuses me `RenderPageForDevelopers(w http.ResponseWriter, r *http.Request, devRepo *developer.Repository, location, tag, page, htmlView string)`, mainly because is a Server method but it's being called by the `DevelopersHandler` handler func

Reply:
yes you can leave that there for now, but it might need to be extracted into a separate function/helper that can be injected as well

Reply:
it's ok to pass it like you did now, we dont have to fix everything at once :)